core/utils/poly-look/src/visualisations/charts/bar-charts/horizontalBarChart/horizontalBarChart.js

```diff
@@ -0,0 +1,329 @@
+import * as d3 from "d3";
+
+import { Chart } from "../../chart";
+
+import "./horizontalBarChart.css";
+const initializingBarHeight = 2;
+const margin = {
+  top: 10,
+  right: 0,
+  bottom: 20,
+  left: 0,
+};
+const barValueMargin = 4;
+const barPaddingProportion = 0.2;
+const groupHeadlineHeight = 14;
+const headlinePadding = 6;
+const barTextBottomMargin = 6;
+const barLabelTopMargin = 2;
+const defaultBarColor = "blue";
+const defaultBarValueColor = "white";
+const defaultBarWidth = 20;
+/**
+ *
+ * @class
+ * @extends Chart
+ * @param {CSS-selector} selector - A CSS selector, where the svg will be attached to
+ * @param {Object[]} data - The data to be visualized as a bubble cluster
+ * @param {string} data[].title - The title/name the bubble has
+ * @param {number} data[].value - The value of the bubble, which corresponds to it's radius
+ * @param {number} [width  = 400] - The width of the svg
+ * @param {number} [height = 400] - The height of the svg
+ * @param {string|callback} [barColor = defaultBarColor] - The color of the bar (callbacks receive event and data)
+ * @param {string} [barValueColor] - The color the values are shown in (default = no values shown)
+ * @param {string} [barWidth] - barWidth is actually bar height since it is a horizontal bar chart but for all intents and purposes it is the bar's width
+ * @param {Object[]} [groups] - This holds the information of what groups are associated with the data (default: dafta is not grouped)
+ */
+
+class ScaleContainer {
+  constructor({ scale, id }) {
+    this.scale = scale;
+    this.id = id;
+  }
+}
+
+export class HorizontalBarChart extends Chart {
+  constructor({
+    selector,
+    data,
+    barWidth,
+    groups,
+    barColor,
+    barValueColor,
+    width = 400,
+    height = 400,
+  }) {
+    super({ selector, data, width, height, margin });
+    this._groups = groups;
+    this._barColor = barColor || defaultBarColor;
+    this._barWidth = defaultBarWidth || barWidth;
+    this._xScale = d3.scaleLinear().range([0, this.chartWidth]);
+    this._yScales = this._getYscales(groups);
+    this._barValueColor = barValueColor || defaultBarValueColor;
+  }
+
+  _getYscales(groups) {
+    if (groups) {
+      const yScales = [];
+      const headingSpace = groupHeadlineHeight + headlinePadding;
+      let headingMargin = headingSpace;
+      for (let group of groups) {
+        const relevantBars = this.data.filter(
+          (data) => data.group === group.id
+        ).length;
+        const step = this._barWidth / (1 - barPaddingProportion);
+        const absolutePad = step * barPaddingProportion;
+        const range = relevantBars * step + absolutePad;
+        const scale = d3
+          .scaleBand()
+          .range([headingMargin, headingMargin + range])
+          .padding(barPaddingProportion);
```

Comment:
We're setting a padding of `0.2` here (and some places below) - does that actually work? AFAIK, those values are in pixels, so this would probably get rounded (`floor`, I believe), and end up being `padding(0)`? Maybe it makes sense, just doesn't look like it does to me :D